File: eva-core/eva-core-upload/src/main/java/io/nerv/core/upload/condition/FastDfsCondition.java
package io.nerv.core.upload.condition;

import org.springframework.context.annotation.Condition;
import org.springframework.context.annotation.ConditionContext;
import org.springframework.core.type.AnnotatedTypeMetadata;

/**
 * 文件上传启用条件， 是否启用go-fastdfs
 *
 * @author: S.PKAQ
 */
public class FastDfsCondition implements Condition {
    @Override
    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
        String cacheType = context.getEnvironment().getProperty("eva.upload.type");
        return "dfs".equals(cacheType);
    }
}

File: eva-core/src/main/kotlin/io/nerv/core/upload/condition/FastDfsCondition.kt
package io.nerv.core.upload.condition

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.core.type.AnnotatedTypeMetadata

/**
 * 文件上传启用条件， 是否启用go-fastdfs
 * @author: S.PKAQ
 */
class FastDfsCondition : Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        val cacheType = context.environment.getProperty("eva.upload.type")
        return "dfs" == cacheType
    }
}